src/main/java/seedu/parser/AddPersonalContactParser.java

```diff
@@ -1,36 +1,167 @@
 package seedu.parser;
 
-
 import seedu.contact.Contact;
+import seedu.exception.InvalidGithubUsernameException;
 import seedu.exception.InvalidNameException;
+import seedu.exception.InvalidTelegramUsernameException;
+import seedu.exception.InvalidTwitterUsernameException;
+import seedu.exception.InvalidLinkedinUsernameException;
+import seedu.exception.InvalidEmailException;
 import seedu.ui.ExceptionTextUi;
 import seedu.ui.TextUi;
 import seedu.ui.UserInputTextUi;
 
 public class AddPersonalContactParser extends RegexParser {
     private Contact personalContact = new Contact(null, null,null,null,
             null,null);
+    private boolean isValidDetail = false;
 
     public AddPersonalContactParser() {
         TextUi.welcomeMessage();
-        promptPersonalName();
+    }
+
+    public void startCollectingPersonalDetails() {
+        parsePersonalName();
         TextUi.greetingMessage(personalContact);
+        String userConfirmation = UserInputTextUi.getUserConfirmation();
+        if (userConfirmation != null || userConfirmation.isBlank()) {
+            parsePersonalGithubUsername();
+            parsePersonalTelegramUsername();
+            parsePersonalTwitterUsername();
+            parsePersonalEmailAddress();
+            parsePersonalLinkedInUsername();
+            TextUi.finishSetUpMessage();
+        }
     }
 
-    private void promptPersonalName() {
-        boolean isValid = false;
-        String personalName = "";
+    private void parsePersonalName() {
+        isValidDetail = false;
         do {
             try {
-                personalName = UserInputTextUi.getUserInput();
+                String personalName = UserInputTextUi.getUserInput();
                 checkNameRegex(personalName);
                 //checkNameRegex passes so the name must be valid
-                isValid = true;
+                isValidDetail = true;
+                this.personalContact.setName(personalName);
             } catch (InvalidNameException e) {
                 ExceptionTextUi.invalidPersonalNameErrorMessage();
             }
-        } while (!isValid);
-        this.personalContact.setName(personalName);
+        } while (!isValidDetail);
+    }
+
+    private void parsePersonalGithubUsername() {
+        isValidDetail = false;
+        TextUi.promptPersonalGithubUsernameMessage(this.personalContact.getName());
+        do {
+            try {
+                String personalGithubUsername = UserInputTextUi.getUserInput();
+                setGithubIfValid(personalGithubUsername);
+            } catch (InvalidGithubUsernameException e) {
+                ExceptionTextUi.invalidPersonalGithubUsernameErrorMessage();
+            }
```

Comment:
Could this be SLAP-ed into another function with return type bool within the while loop?